browser/app/js/actions.js

```diff
@@ -314,6 +337,37 @@ export const selectBucket = (newCurrentBucket, prefix) => {
   }
 }
 
+export const deleteBucket = (bucket) => {
+  return (dispatch, getState) => {
+    // DeleteBucket() RPC call will ONLY delete a bucket if it is empty of
+    // objects. This means a call can just be sent, as it is entirely reversable
+    // and won't do any permanent damage.
+    web.DeleteBucket({
+      bucketName: bucket
+    })
+      .then(() => {
+        dispatch(showAlert({
+          type: 'info',
+          message: `Bucket '${bucket}' has been deleted.`
+        }))
+        dispatch(removeBucket(bucket))
+      })
+      .catch(err => {
+        let message = err.message
+
+        // Show a custom "bucket not empty" message, as it can be confusing.
+        if (/Bucket not empty/.test(err.message)) {
```

Comment:
I think we don't need to do this for the moment, we can show `Bucket not empty` and let's care about UI error messages in a separate PR with a generic solution.

Reply:
So I should remove the `if` statement with the `test()` method and just keep the error message as `Bucket '${bucket}' must be empty to delete.`, right?